src/main/java/tool/designpatterns/verifiers/VerifierUtils.java

```diff
@@ -141,4 +141,41 @@ public static Feedback hasFieldOfType(
         return result;
     }
 
+    /**
+     * If a method has the {@link Override} Annotation we know that it is implemented. And if it
+     * then has the same name and argumnents, it has the same method head and is therefore the same
+     * method.
+     *
+     * @param method    methods to check if it can come from the component
+     * @param component the class or interface with implementable methods
+     *
+     * @return a boolean specifying if the method can come from the component
+     */
+    public static boolean methodBelongsToComponent(
+        MethodDeclaration method, ClassOrInterfaceDeclaration component) {
+        for (MethodDeclaration methodInContainer : component.getMethods()) {
+            Optional<AnnotationExpr> overrideAnn = method.getAnnotationByClass(Override.class);
+            if (isSameMethod(methodInContainer, method) && overrideAnn.isPresent()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * As getQualifiedName returns the package path to the method, it cant be used to determine if
+     * two merhods are the same. Instead, we check that they have the same name and idendentical
+     * argments.
+     *
+     * @param method1 a method
+     * @param method2 a method
+     *
+     * @return a boolean
+     */
+    public static boolean isSameMethod(MethodDeclaration method1, MethodDeclaration method2) {
+        boolean hasSameName = method1.getName().equals(method2.getName());
+        boolean hasSameParameters = method1.getParameters().equals(method2.getParameters());
+        return hasSameName && hasSameParameters;
```

Comment:
Shouldn't this be done with the symbolsolver?

Reply:
```suggestion
        String qualifiedName1 = method1.resolve().getQualifiedName();
        String qualifiedName2 = method2.resolve().getQualifiedName();
        return qualifiedName1.equals(qualifiedName2);
```

Reply:
no is wrong. But is change name